cadence/contracts/KittyItems.cdc

```diff
@@ -270,6 +271,21 @@ pub contract KittyItems: NonFungibleToken {
 
             KittyItems.totalSupply = KittyItems.totalSupply + (1 as UInt64)
         }
+
+        // Update NFT images for new type
+        pub fun addNewImagesForKind(from: AuthAccount, newImages: {Kind: {Rarity: String}}) {
+            let kindValue = KittyItems.images.containsKey(newImages.keys[0]) 
+            if(!kindValue) {
+                KittyItems.images.insert(key: newImages.keys[0], newImages.values[0])
+                emit ImagesAddedForNewKind(
+                    kind: newImages.keys[0],
+                )
+            } else {
+                panic("No Rugs... Can't update existing NFT images.")
```

Comment:
Not sure if we want to allow this or not. 

If we update the images for an existing kind, then all subsequent mints would have the new images, but old mints (of the same kind) would have the old images. 🤷🏼